opensrp-anc/src/main/java/org/smartregister/anc/library/widget/ANCSpinnerFactory.java

```diff
@@ -0,0 +1,192 @@
+        {
+            put(PROVINCE, Arrays.asList(DISTRICT, SUB_DISTRICT, FACILITY, VILLAGE));
+            put(DISTRICT, Arrays.asList(SUB_DISTRICT, FACILITY, VILLAGE));
+            put(SUB_DISTRICT, Arrays.asList(FACILITY, VILLAGE));
+            put(FACILITY, Arrays.asList(VILLAGE));
+            put(VILLAGE, null);
+        }
+    };
+
+    private final Map<String, String> parents = new HashMap<String, String>() {
+        {
+            put(DISTRICT, PROVINCE);
+            put(SUB_DISTRICT, DISTRICT);
+            put(FACILITY, SUB_DISTRICT);
+            put(VILLAGE, FACILITY);
+        }
+    };
+    private JsonFormFragment formFragment;
+    private JsonFormActivity jsonFormView;
+
+    @Override
+    public void genericWidgetLayoutHookback(View view, JSONObject jsonObject, JsonFormFragment formFragment) {
+        super.genericWidgetLayoutHookback(view, jsonObject, formFragment);
+        View materialView = ((RelativeLayout) view).getChildAt(0);
+        materialView.setOnTouchListener((v, event) -> {
+            materialView.setTag(R.id.is_human_action, true);
+            return false;
+        });
+    }
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment,
+                                       JSONObject jsonObject, CommonListener listener, boolean popup) throws Exception {
+        this.formFragment = formFragment;
+        jsonFormView = (JsonFormActivity) formFragment.getActivity();
+
+        try {
+            if (jsonObject.has(SUB_TYPE)
+                    && jsonObject.getString(SUB_TYPE)
+                    .equalsIgnoreCase(LOCATION_SUB_TYPE)
+                    && jsonObject.has(ConstantsUtils.JsonFormKeyUtils.OPTIONS)
+                    && jsonObject.getJSONArray(ConstantsUtils.JsonFormKeyUtils.OPTIONS).length() <= 0) {
+
+                String stepTitle = formFragment.getStep(STEP1).get(STEP_TITLE).toString();
+                if (stepTitle.equals(ConstantsUtils.EventTypeUtils.REGISTRATION)
+                        && StringUtils.endsWithIgnoreCase(jsonObject.getString(KEY), "province")) {
+                    populateProvince(jsonObject);
+                } else {
+                    populateDescendants(jsonObject);
+                }
+            }
+        } catch (JSONException e) {
+            Timber.e(e);
+        }
+
+        return super.getViewsFromJson(stepName, context, formFragment, jsonObject, listener, popup);
+    }
+
+    protected void populateProvince(JSONObject jsonObject) {
+        List<LocationTag> tags = Utils.getLocationTagsByTagName("Country");
+        // TODO: Remove static country ID, current implementation is only done to select proper country in case of more than one countries (dummy data) in locations
+        String countryId = (tags != null && tags.size() > 0) ? ((tags.size() > 1) ? "02ebbc84-5e29-4cd5-9b79-c594058923e9" : tags.get(0).getLocationId()) : "";
```

Comment:
Please remember to mos the static country ID to use configs from the `app.properties`. also, remember to check when it's set and log any errors appropriately

Reply:
Country ID is now referenced from `app.properties`